src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -92,15 +111,41 @@ public class WwwAuthenticateParameters
         {
             if (httpResponseHeaders.WwwAuthenticate.Any())
             {
-                // TODO: add POP support
-                AuthenticationHeaderValue bearer = httpResponseHeaders.WwwAuthenticate.First(v => string.Equals(v.Scheme, scheme, StringComparison.OrdinalIgnoreCase));
+                AuthenticationHeaderValue bearer = httpResponseHeaders.WwwAuthenticate.First(v => string.Equals(v.Scheme, Constants.BearerAuthHeaderPrefix, StringComparison.OrdinalIgnoreCase));
                 string wwwAuthenticateValue = bearer.Parameter;
-                return CreateFromWwwAuthenticateHeaderValue(wwwAuthenticateValue);
+                var parameters = CreateFromWwwAuthenticateHeaderValue(wwwAuthenticateValue);
+                parameters.AuthScheme = Constants.BearerAuthHeaderPrefix;
+                return parameters;
             }
 
             return CreateWwwAuthenticateParameters(new Dictionary<string, string>());
         }
 
+        /// <summary>
+        /// Create WWW-Authenticate parameters from the HttpResponseHeaders for each auth scheme.
+        /// </summary>
+        /// <param name="httpResponseHeaders">HttpResponseHeaders.</param>
+        /// <returns>The parameters requested by the web API.</returns>
+        /// <remarks>Currently it only supports the Bearer scheme</remarks>
+        public static IEnumerable<WwwAuthenticateParameters> CreateAllFromResponseHeaders(
```

Comment:
how about we return `IReadOnlyList` ?

Reply:
Actually it looks like the .NET guidelines provide an answer here: 

- use IEnumerable<T> for input params
- use ReadOnlyCollection<T> for return types 

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections?redirectedfrom=MSDN - very good and short read (CC @pmaytak , @jmprieur, @gladjohn, @neha-bhargava, @SameerK-MSFT  )